mountinfo/mountinfo_unix.go

```diff
@@ -0,0 +1,37 @@
+// +build !windows
+
+package mountinfo
+
+import (
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+// Mounted determines if path is mounted (i.e. it is a mount point).
+func Mounted(path string) (bool, error) {
+	// Try the fast method (of comparing devices) first.
+	// Alas, it can't detect bind mounts, so fall back on
+	// good ol' mountinfo parsing.
+	var st, pst unix.Stat_t
+
+	err := unix.Lstat(path, &st)
```

Comment:
Just noticed; `unix.Lstat` is not available on macOS. While this function was never functional on macOS, it looks like it did _compile_ before (but called out to `parseMountTable()`,  which returned an error)
https://github.com/moby/sys/blob/0783079dd297647b492eef4d729d657cc7d3bc6c/mountinfo/mountinfo_unsupported.go#L11-L13

Same for freebsd https://github.com/moby/sys/blob/0783079dd297647b492eef4d729d657cc7d3bc6c/mountinfo/mountinfo_unsupported.go#L1

Reply:
AFAIK Darwin does have lstat(2), for example, see

https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/stat.2.html

Have you tried it?

Reply:
Arf, looks like my go modules cache was probably out of date 😞

I do see that the build tags need updating in your latest version; for `darwin`, `mounted()` is now "redeclared"

Reply:
the _unsupported.go has `// +build !windows,!linux,!freebsd freebsd,!cgo`, so should probably have `!darwin` added, but doing so would then make `parseMountTable` and `parseInfoFile` go missing on `darwin` (because they're in the `_linux.go` file 😞